### PR TITLE
feat: Use a new way to keep Kahla up-to-date

### PR DESCRIPTION
**Pull request by hv0905**
Warning: this commit haven't been tested deeply.

### CODE REVIEW

**Comment by Anduin2017** [comment]
This is a really nice fix. But how can we test it?

File: src/app/Services/CheckService.ts [hunk 1]
```diff
@@ -57,8 +78,18 @@ export class CheckService {
                     this.openWebPage(downloadAddress);
                 }
             });
+        } else if (this.checkSwCache()) {
+            this.updateServiceWorkerCache();
+            if (showAlert) {
+                Swal.fire({
+                    title: 'There is a new version of Kahla!',
+                    text: 'We are upgrading automatically at the background.\n You will be notified when done.',
```
Comment (Anduin2017) [inline src/app/Services/CheckService.ts:86]: It seems that sending a new alert is too annoying. What about just display a yellow bar if the patch number is different. And show alert only when the major version number is different.

Comment (hv0905) [reply src/app/Services/CheckService.ts:86]: This alert will only be shown when user manually click "check update" button.

Comment (Anduin2017) [reply src/app/Services/CheckService.ts:86]: Ohh, that's fine.

**Review by Anduin2017 — APPROVED** [review]

**Pull request merged**